python/cudf/cudf/core/series.py

```diff
@@ -4503,6 +4626,48 @@ def keys(self):
         """
         return self.index
 
+    def _drop_rows_by_labels(
+        self, labels: ColumnLike, level: Union[int, __builtins__.str] = None
+    ) -> "cudf.Series":
+        """Delete rows specified by `label` parameter. Resort to the efficient
+        implementation in `cudf.DataFrame`
+
+        labels: a list of labels specifying the rows to drop
+        """
+
+        if isinstance(self._index, cudf.MultiIndex):
+            if isinstance(level, int):
+                ilevel = level
+            else:
+                ilevel = self._index.names.index(level)
+
+            idx_nlv = self._index.nlevels
+            working_df = self._index._source_data
+            # TODO: figure out what __unique__ should be
+            working_df["__unique__"] = self._column
```

Comment:
I'm using kind of an awkward way to combine two dataframes together, any idea how to improve?

Reply:
Can we instead just construct `working_df` with column labels `0, 1, ... n` where the first (or last) `m` columns represent the index?